### PR TITLE
feat(form-v2): attachment component

### LINKED ISSUE #2010: [React] Add Field/AttachmentField component
**Time estimation: 1w**

File handling state logic might to be implemented to display various states. Check out other component libraries for how they expose the various props for logic

E.g. https://5d559397bae39100201eedc1-fdkpcaaxrr.chromatic.com/?path=/story/all-components-drop-zone--drop-zone-with-file-upload

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
We don't have an attachment component and this PR adds that in. This PR uses the compound component pattern together with custom hooks to achieve that.

Closes #2010 

## Solution
This solution is comprised of 3 main sections. 

**useAttachments hook** 
This is a custom hook written to abstract over `useDropzone` - the reason for doing so is to add additional functionality such as resetting and deleting files which `useDropzone` does not provide for. 

Additionally, having the custom hook written allows for us to put our stateful logic into a single place and allows for cleaner UI components.

**useAttachmentContext hook**
This custom hook was written to have props shared across the `Attachment` component. This is because we are writing a design system and our components should be easily composable, they require certain props (eg: the # of acceptedFiles, the # of rejections) hidden away to allow for ease of use. 

The hook and the context hides shared props, allowing our components to still have access to them while allowing composability

**compound components**
The `Attachment` component is made up of numerous smaller components, which can be easily used in tandem or in isolation. 

As they are only loosely coupled and the base components only have minor styling, this allows for users to use different parts of the `Attachment` component without being too coupled to its styling. 

**New dependencies**:

- `react-dropzone`

## Accessibility Tests 
- [ ] Screen readers automatically read out file description after successfully uploading 
- [ ] Delete button is tabbable 
- [ ] Screen readers read out error when an error does occur

## Screenshots
available in generated storybook

### CODE REVIEW

**Comment by karrui** [comment]
Should we also have a story for the active state?

Something like https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader-item

**Comment by seaerchin** [comment]
transcribing from discussion: 

overeager adoption of compound components pattern led to 2 problems: 
1. the component is overly complicated because it exposed all of its internals as subcomponents. this shouldn't be the case; rather, we should expose only necessary wrappers so that the complexity is managed well.
2. the component is annoying to use because we'd have to compose it ourselves, due to the lack of a default export. this increases dev friction.

while this component is composable, most of the time people will not lead the extra stylability/customizability (those can instead be done using the subcomponents) and we should build for the common use case of other devs using our component as-is

File: frontend/src/components/Field/Attachment/Attachment.tsx [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    )}`}</FormFieldMessage>
+  )
+}
+
+export interface AttachmentLabelProps {
+  number?: number
+  title?: string
+}
+
+export const AttachmentLabel = ({
```
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/components/Field/Attachment/Attachment.tsx:321]: As discussed on Slack, this component should only be responsible for rendering the attachment field and nothing else~

**Comment by karrui** [comment]
Superseded by https://github.com/opengovsg/FormSG/pull/2649.

**Pull request closed without merging**